slatedb/src/compactor_executor.rs

```diff
@@ -36,6 +37,7 @@ pub(crate) struct CompactionJob {
     pub(crate) compaction_ts: i64,
     pub(crate) is_dest_last_run: bool,
     pub(crate) retention_min_seq: Option<u64>,
+    pub(crate) sequence_tracker: Arc<SequenceTracker>,
```

Comment:
Hmm. It feels off to put the `sequence_tracker` here. I view the `CompactionJob` as a declarative, serializable structure. This is further reinforced in @sujeetsawala's on-going work on persistent state. Can we instead supply the sequence_tracker to `TokioCompactionExecutor::new`?